salt/utils/network.py

```diff
@@ -1881,7 +1881,12 @@ def dns_check(addr, port=80, safe=False, ipv6=None):
                     resolved = salt.utils.zeromq.ip_bracket(addr)
                     break
 
-                candidate_addr = salt.utils.zeromq.ip_bracket(h[4][0])
+                #Don't use brackets around interface if using tcp transport
+                if transport == 'tcp':
+                    candidate_addr = h[4][0]
+                else:
+                    candidate_addr = salt.utils.zeromq.ip_bracket(h[4][0])
```

Comment:
You can use ternary operator as well instead of all this thing:
```python
candidates.append(h[4][0] if transport == 'tcp' else salt.utils.zeromq.ip_bracket(h[4][0]))
```